pkg/prometheus/promcfg.go

```diff
@@ -4509,6 +4515,18 @@ func (cg *ConfigGenerator) appendOTLPConfig(cfg yaml.MapSlice) (yaml.MapSlice, e
 		}), nil
 }
 
+func (cg *ConfigGenerator) appendNameValidationScheme(cfg yaml.MapSlice) yaml.MapSlice {
+	nameValScheme := cg.prom.GetCommonPrometheusFields().NameValidationScheme
+
+	if nameValScheme == "" {
+		return cfg
+	}
+	if nameValScheme != "utf8" && nameValScheme != "legacy" {
+		return cfg
+	}
+	return cg.WithMinimumVersion("v3.0.0-beta.0").AppendMapItem(cfg, "metric_name_validation_scheme", nameValScheme)
```

Comment:
```suggestion
	if nameValScheme != nil && cg.version.Major >= 3 {
		return cg.AppendMapItem(cfg, "metric_name_validation_scheme", nameValScheme)
	}
	return cfg
```

Reply:
Well since the change from https://github.com/prometheus/prometheus/pull/14705 only appears to be in [v3.0.0-beta.0](https://github.com/prometheus/prometheus/releases/tag/v3.0.0-beta.0) I think I need to check for that specific version and not 3 overall, WTDY?

Reply:
Ok thans, added the fix, pls review